### DISCUSSION TITLE
Store values in session to persist across page loads?

### DISCUSSION THREAD

**Discussion by onionjake** [jfyne/live#31] [Q&A]
In `session.go` a session is declared as this:

```
// Session what we will actually store across page loads.
type Session struct {
	ID string
}
```

Is there already a way to store additional values other than just the `ID` that will persist across page loads?

I noticed there is also this type defined:

```
// ValueKey type for session keys.
type ValueKey string
```

But it doesn't seem to be used anywhere...

**Comment by jfyne** [comment]
Hi, can you tell me what you are looking to store in there?

This part of the code needs another look at, and I would be happy to see if storing arbitrary info in there could be done.

**Reply by onionjake** [reply]
Sure. I was doing a simple login and was hoping to store some information about the logged in user.

**Reply by jfyne** [reply]
Can you have a look at: https://github.com/jfyne/live/pull/32

**Reply by jfyne** [reply]
As a side note, what I have done in our projects is just use middleware and a different cookie to store the user data

**Reply by onionjake** [reply]
Oh middleware would have worked too. I just saw there was a cookie there already and wanted to use it.
